src/Client.ts

```diff
@@ -716,19 +716,24 @@ export class Client {
    * @returns {void|Promise} void if not connected; otherwise returns a promise to the request to disconnect
    */
   public async unbind(): Promise<void> {
-    if (!this.connected || !this.socket) {
-      return;
-    }
+    try {
+      if (!this.connected || !this.socket) {
+        return;
+      }
 
-    const req = new UnbindRequest({
-      messageId: this._nextMessageId(),
-    });
+      const req = new UnbindRequest({
+        messageId: this._nextMessageId(),
+      });
 
-    await this._send(req);
+      await this._send(req);
+    } finally {
+      this._destroySocket(this.socket);
+    }
   }
 
-  public [Symbol.asyncDispose](): Promise<void> {
-    return this.unbind();
+  public async [Symbol.asyncDispose](): Promise<void> {
+    await this.unbind().catch();
+    this._destroySocket(this.socket);
```

Comment:
The change to`[Symbol.asyncDispose]()` seems unnecessary since `unbind()` will destroy the socket.